frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/block-pool.ts

```diff
@@ -0,0 +1,101 @@
+import { NS } from '../utils/consts';
+import { commonFlows } from './common';
+import { POOL_PROGRESS } from '../../src/constants/storage-pool-const';
+
+// Pool var
+export const poolName: string = 'example.pool';
+export const replicaCount: string = '2';
+export const volumeType: string = 'ssd';
+
+export const poolMessage: {
+  [key in POOL_PROGRESS]?: string;
+} = {
+  [POOL_PROGRESS.FAILED]: `Pool "${poolName}" already exists`,
+  [POOL_PROGRESS.CREATED]: `Pool ${poolName} was successfully created`,
+  [POOL_PROGRESS.NOTALLOWED]:
+    "Pool management tasks are not supported for default pool and Openshift Container Storage's external mode.",
+};
+
+export const navigateToBlockPool = () => {
+  commonFlows.navigateToOCS();
+  cy.byLegacyTestID('horizontal-link-Block Pools').click();
+};
+
+export const populateBlockPoolform = () => {
+  cy.byTestID('new-pool-name-textbox').type(poolName);
+  cy.byTestID('replica-dropdown').click();
+  cy.byLegacyTestID(replicaCount)
+    .last()
+    .contains(`${replicaCount}-way Replication`)
+    .click();
+  cy.byTestID('volume-type-dropdown').click();
+  cy.byTestID(volumeType).click();
+  cy.byTestID('compression-checkbox').check();
+};
+
+export const blockPoolFooter = (action: string, visible: boolean = false) => {
+  switch (action) {
+    case 'failed':
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.FAILED]);
+      cy.byTestID('modal-try-again-action').should('be.visible');
+      cy.byTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
+      break;
+    case 'created':
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.CREATED]);
+      cy.byTestID('modal-finish-action')
+        .should('be.visible')
+        .click();
+      break;
+    case 'notAllowed':
+      cy.byTestID('empty-state-body').contains(poolMessage[POOL_PROGRESS.NOTALLOWED]);
+      cy.byTestID('modal-close-action')
+        .should('be.visible')
+        .click();
+      break;
+    default:
+      cy.byTestID('cancel-action')
+        .scrollIntoView()
+        .should('be.visible');
+      visible
+        ? cy
+            .byTestID('confirm-action')
+            .should('be.visible')
+            .click()
+        : cy.byTestID('confirm-action').should('be.disabled');
+  }
+};
+
+export const verifyBlockPool = (
+  compressionEnabled: boolean = true,
+  replica: string = replicaCount,
+) =>
+  cy.exec(`oc get cephBlockPool ${poolName} -n openshift-storage -o json`).then((res) => {
+    const blockPool = JSON.parse(res.stdout);
+    expect(blockPool.spec?.replicated?.size).toEqual(Number(replica));
+    expect(blockPool.spec?.compressionMode).toEqual(compressionEnabled ? 'aggressive' : '');
+    expect(blockPool.spec?.parameters?.compression_mode).toEqual(
+      compressionEnabled ? 'aggressive' : '',
+    );
+  });
+
+export const createBlockPool = (alreadyExists: boolean = false) => {
```

Comment:
create block pool from block pool page, and it will be re-used by update and delete testcases